app/src/main/java/com/conupods/IndoorMaps/IndoorBuildingOverlays.java

```diff
@@ -0,0 +1,238 @@
+package com.conupods.IndoorMaps;
+
+import android.os.Build;
+import android.util.Log;
+import android.view.View;
+
+import com.conupods.OutdoorMaps.Building;
+import com.conupods.OutdoorMaps.BuildingDataMap;
+import com.conupods.R;
+import com.google.android.gms.maps.GoogleMap;
+import com.google.android.gms.maps.model.BitmapDescriptor;
+import com.google.android.gms.maps.model.BitmapDescriptorFactory;
+import com.google.android.gms.maps.model.GroundOverlay;
+import com.google.android.gms.maps.model.GroundOverlayOptions;
+import com.google.android.gms.maps.model.LatLng;
+import com.google.android.gms.maps.model.MapStyleOptions;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class IndoorBuildingOverlays {
+
+    private GoogleMap mMap;
+
+    public enum Buildings{
+        HALL, MB, VL, CC
+    }
+    
+    private static final LatLng Building_MB = new LatLng(45.49575150228435, -73.5789343714714);
+    private static final LatLng Building_LOY_CC = new LatLng(45.45863873466155,  -73.64075660705566);
+    private static final LatLng Building_LOY_VL = new LatLng(45.45890400660071,   -73.63919287919998);
+    private static final LatLng Building_HALL = new LatLng(45.497273, -73.578955);
+    private static final LatLng NEAR_Building_HALL = new LatLng(Building_HALL.latitude + 0.0005, Building_HALL.longitude - 0.0001);
+
+
+    private List<BitmapDescriptor> mImages = new ArrayList<BitmapDescriptor>();
+    private GroundOverlay mHALLOverlay;
+    private List<GroundOverlay> Overlays = new ArrayList<GroundOverlay>();
+    private GroundOverlay mMBOverlay;
+    private GroundOverlay mLOYCCOverlay;
+    private GroundOverlay mLOYVLOverlay;
+    private View mLevelButtons;
+    private View floorButtonsHall;
+    private View floorButtonsMB;
+    private View floorButtonsLOY_VL;
+
+
+
+
+    public IndoorBuildingOverlays(View LevelButtons, GoogleMap map) {
+        mMap = map;
+        mLevelButtons = LevelButtons;
+        mMap.setIndoorEnabled(false);
+
+        mImages.clear();
+
+        //index = 0 is first floor of Hall
+        mImages.add(BitmapDescriptorFactory.fromResource(R.drawable.h1));
+        mImages.add(BitmapDescriptorFactory.fromResource(R.drawable.h2));
+        mImages.add(BitmapDescriptorFactory.fromResource(R.drawable.h8));
+        mImages.add(BitmapDescriptorFactory.fromResource(R.drawable.h9));
+
+        //index = 4 is first floor of MB
+        mImages.add(BitmapDescriptorFactory.fromResource(R.drawable.mb1));
+        mImages.add(BitmapDescriptorFactory.fromResource(R.drawable.mbs2));
+
+        //index = 6 is first floor of LOY_CC
+        mImages.add(BitmapDescriptorFactory.fromResource(R.drawable.cc1));
+
+        //index = 7 is first floor of LOY_VL
+        mImages.add(BitmapDescriptorFactory.fromResource((R.drawable.loy_vl1)));
+        mImages.add(BitmapDescriptorFactory.fromResource(R.drawable.loy_vl2));
```

Comment:
All the increment in index correspond into an increment in floor level, but for MB, where index 2 is S2. I believe this could cause issues, especially if someones writes an enum function to go over them. I would strongly advise to think about something more consistent.

Reply:
I do agree that it's inconsistent. Would it be better to have an array for each building.

Reply:
For now we will have to think about how to implement this so it doesn't become a roadblock for you in the future.